src/SourceEngine.Demo.Parser/DemoParser.cs

```diff
@@ -552,6 +557,25 @@ public DemoParser(Stream input, bool parseChickens = true, bool parsePlayerPosit
 
             this.parseChickens = parseChickens;
             this.parsePlayerPositions = parsePlayerPositions;
+			this.gamemode = gamemode;
+
+			// automatically decides rescue zone amounts unless overridden with a provided parameter
+			if (hostagerescuezonecountoverride > 0)
+			{
+				this.numOfHostageRescueZonesLookingFor = hostagerescuezonecountoverride;
+			}
+			else if (gamemode == Gamemodes.DangerZone)
+			{
+				this.numOfHostageRescueZonesLookingFor = 2;
+			}
+			else if (gamemode == Gamemodes.Hostage)
+			{
+				this.numOfHostageRescueZonesLookingFor = 1;
+			}
+			else
+			{
+				this.numOfHostageRescueZonesLookingFor = 0;
+			}
```

Comment:
This feels hacky. Did reading it as an array not work? There must be a better way. At the least, I know the number of elements in the array is networked (but not necessarily how to retrieve that information).

Reply:
It can probably be refactored, as it would be better if hostagerescuezonecountoverride and numOfHostageRescueZonesLookingFor were completely removed and unnecessary to have.

Ideally it will just grab all hostage rescue zones that are in the map.

These seem to be called multiple times for each hostage rescue zone with different values. It seems that the last values it receives are correct (for both rescue zones when parsing a danger zone map), I didn't have chance to look into why this happens, so I ended up doing it this hacky not very nice way.